templates/CMakeLists.txt.template

```diff
@@ -71,14 +71,20 @@
 
   cmake_minimum_required(VERSION 3.5.1)
 
-  set(PACKAGE_NAME      "grpc")
-  set(PACKAGE_VERSION   "${settings.cpp_version}")
-  set(gRPC_CORE_VERSION "${settings.core_version}")
-  set(PACKAGE_STRING    "<%text>${PACKAGE_NAME} ${PACKAGE_VERSION}</%text>")
-  set(PACKAGE_TARNAME   "<%text>${PACKAGE_NAME}-${PACKAGE_VERSION}</%text>")
-  set(PACKAGE_BUGREPORT "https://github.com/grpc/grpc/issues/")
+  set(PACKAGE_NAME        "grpc")
+  set(PACKAGE_VERSION     "${settings.cpp_version}")
+  set(gRPC_CORE_VERSION   "${settings.core_version}")
+  set(gRPC_CPP_VERSION    "${settings.cpp_version}")
+  set(gRPC_CSHARP_VERSION "${settings.csharp_version}")
+  set(PACKAGE_STRING      "<%text>${PACKAGE_NAME} ${PACKAGE_VERSION}</%text>")
+  set(PACKAGE_TARNAME     "<%text>${PACKAGE_NAME}-${PACKAGE_VERSION}</%text>")
+  set(PACKAGE_BUGREPORT   "https://github.com/grpc/grpc/issues/")
   project(<%text>${PACKAGE_NAME}</%text> LANGUAGES C CXX)
 
+  string(REGEX REPLACE "^([0-9]+)\\..*$" "\\1" gRPC_CORE_SOVERSION "<%text>${gRPC_CORE_VERSION}</%text>")
```

Comment:
it's not immediately obvious what it does. If you're looking to extract the major version only, I think settings.cpp_version.major should just work.

See https://github.com/grpc/grpc/blob/master/tools/buildgen/plugins/expand_version.py

Reply:
Indeed. Fixed.